yaml2workflow/parser.py

```diff
@@ -54,26 +56,55 @@ def parse(filename: str, generate_new_id: bool = False, stub: service_pb2_grpc.V
     return [resources_pb2.Workflow(id=workflow_id, nodes=nodes)]
 
 
-def parse_model(model: typing.Dict, stub: service_pb2_grpc.V2Stub = None,
-                metadata=None) -> resources_pb2.Model:
-    model_id, model_version_id = search_model(model, stub=stub, metadata=metadata)
+def _parse_model(yaml_model: typing.Dict, stub: service_pb2_grpc.V2Stub = None,
+                 metadata=None) -> resources_pb2.Model:
+    model_id, model_version_id = _search_model(yaml_model, stub=stub, metadata=metadata)
 
     return resources_pb2.Model(
         id=model_id,
         model_version=resources_pb2.ModelVersion(id=model_version_id))
 
 
-def search_model(model: typing.Dict, stub: service_pb2_grpc.V2Stub = None,
-                 metadata=None) -> typing.Tuple[str, str]:
-    if 'model_id' in model and 'model_version_id' in model:
-        return model['model_id'], model['model_version_id']
+def _search_model(yaml_model: typing.Dict, stub: service_pb2_grpc.V2Stub = None,
+                  metadata=None) -> typing.Tuple[str, str]:
+    assert stub, "Stub is required to load model by ID."
+    assert metadata, "Metadata is required to load model by ID."
+    model_version_id = yaml_model['model_version_id'] if 'model_version_id' in yaml_model else None
+    response = stub.GetModel(service_pb2.GetModelRequest(model_id=yaml_model['model_id'], version_id=model_version_id),
+                             metadata=metadata)
+    if response.status.code == status_code_pb2.MODEL_DOES_NOT_EXIST:
+        raise NotImplementedError("Should create new model %s" % yaml_model)
 
-    if 'model_id' in model:
-        assert stub, "Stub is required to load model by ID."
-        assert metadata, "Metadata is required to load model by ID."
-        response = stub.GetModel(service_pb2.GetModelRequest(model_id=model['model_id']), metadata=metadata)
-        assert response.status.code == 10000, f'Invalid response {response}'
+    assert response.status.code == status_code_pb2.SUCCESS, f'Invalid response {response}'
 
+    if _is_same_model(response.model, yaml_model):
         return response.model.id, response.model.model_version.id
-
-    raise Exception("Unable to search model for %s" % model)
+    else:
+        if model_version_id:
+            raise Exception("Model version with specific ID '%s' not found for model with ID '%s'." % (
+                model_version_id, yaml_model['model_id']))
```

Comment:
If we are here, then we specified a model version ID in the request, and the response was NOT `MODEL_DOES_NOT_EXIST`, but the model does not match what we have in our YAML. This means that our YAML has both a model version ID and params? That should be allowed.

Reply:
agreed we should not allow both model version ID and params.